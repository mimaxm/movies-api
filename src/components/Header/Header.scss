.navbar {
  padding: 20px;
  ul {
    display: flex;
    justify-content: center;
  }
  li {
    margin-right: 30px;
    &:last-child {
      margin-right: 0;
    }
  }
  &__link {
    text-transform: uppercase;
    letter-spacing: 0.1em;
    transition: all 0.1s;
    &:hover {
      border-bottom: 2px solid;
    }
    &--active {
      font-weight: 700;
      border-bottom: 2px solid;
    }
  }
}
