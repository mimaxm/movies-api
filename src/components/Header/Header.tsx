import "./Header.scss";
import { NavLink } from "react-router-dom";

export const Header = () => {
	return (
		<div>
			<nav className="navbar">
				<ul>
					<li>
						<NavLink to="/" exact className="navbar__link" activeClassName="navbar__link--active">Home</NavLink>
					</li>
					<li>
						<NavLink to="/about" className="navbar__link" activeClassName="navbar__link--active">About</NavLink>
					</li>
				</ul>
			</nav>
		</div>
	);
};
