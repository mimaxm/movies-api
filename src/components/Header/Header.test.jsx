import React from 'react';
import { Header } from './Header';
import { render } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';

test('component should render', () => {
   const component = render(
      <BrowserRouter>
         <Header />
      </BrowserRouter>
   );
   expect(component).toMatchSnapshot();
});
