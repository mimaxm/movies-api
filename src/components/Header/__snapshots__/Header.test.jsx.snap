// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component should render 1`] = `
LoadedCheerio {
  "0": Node {
    "attribs": Object {},
    "children": Array [
      Node {
        "attribs": Object {
          "class": "navbar",
        },
        "children": Array [
          Node {
            "attribs": Object {},
            "children": Array [
              Node {
                "attribs": Object {},
                "children": Array [
                  Node {
                    "attribs": Object {
                      "aria-current": "page",
                      "class": "navbar__link navbar__link--active",
                      "href": "/",
                    },
                    "children": Array [
                      Node {
                        "data": "Home",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "a",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "aria-current": undefined,
                      "class": undefined,
                      "href": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "aria-current": undefined,
                      "class": undefined,
                      "href": undefined,
                    },
                  },
                ],
                "name": "li",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Node {
                  "attribs": Object {},
                  "children": Array [
                    Node {
                      "attribs": Object {
                        "class": "navbar__link",
                        "href": "/about",
                      },
                      "children": Array [
                        Node {
                          "data": "About",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                      ],
                      "name": "a",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "class": undefined,
                        "href": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "class": undefined,
                        "href": undefined,
                      },
                    },
                  ],
                  "name": "li",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {},
                  "x-attribsPrefix": Object {},
                },
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              Node {
                "attribs": Object {},
                "children": Array [
                  Node {
                    "attribs": Object {
                      "class": "navbar__link",
                      "href": "/about",
                    },
                    "children": Array [
                      Node {
                        "data": "About",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "a",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                      "href": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                      "href": undefined,
                    },
                  },
                ],
                "name": "li",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": Node {
                  "attribs": Object {},
                  "children": Array [
                    Node {
                      "attribs": Object {
                        "aria-current": "page",
                        "class": "navbar__link navbar__link--active",
                        "href": "/",
                      },
                      "children": Array [
                        Node {
                          "data": "Home",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                      ],
                      "name": "a",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "aria-current": undefined,
                        "class": undefined,
                        "href": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "aria-current": undefined,
                        "class": undefined,
                        "href": undefined,
                      },
                    },
                  ],
                  "name": "li",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {},
                  "x-attribsPrefix": Object {},
                },
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
            ],
            "name": "ul",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {},
            "x-attribsPrefix": Object {},
          },
        ],
        "name": "nav",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {},
    "x-attribsPrefix": Object {},
  },
  "_root": LoadedCheerio {
    "0": Node {
      "children": Array [
        Node {
          "attribs": Object {},
          "children": Array [
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "head",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "body",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "body",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "head",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
          ],
          "name": "html",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
